lib/data-types.js

```diff
@@ -39,24 +42,47 @@ var util = require('util')
  */
 
 var ABSTRACT = function(options) {
-
 };
 
 ABSTRACT.prototype.dialectTypes = '';
 
-ABSTRACT.prototype.toString = function() {
-  return this.toSql();
+ABSTRACT.prototype.toString = function(options) {
+  return this.toSql(options);
 };
 ABSTRACT.prototype.toSql = function() {
   return this.key;
 };
-ABSTRACT.prototype.warn = function(text) {
-  if (!warnings[text]) {
-    warnings[text] = true;
-    console.warn('>> WARNING:', text, '\n>> Check:', this.dialectTypes);
+ABSTRACT.warn = function(link, text) {
+  deprecate(text + '\n>> Check: ' + link);
```

Comment:
@BridgeAR Yep, you're right.

I just did a check, and everywhere we use depd is actually called from internally (e.g. [here](https://github.com/sequelize/sequelize/blob/master/lib/utils.js#L377)), so it never works as intended :D

I was thinking of writing my own version of depd, which basically traverses up the stack until it meets a filename that's not `node_modules/sequelize/lib/**`. Dunno how hard its going to be, but right now depd is not working any better than what we had before :D